src/gromo/utils/utils.py

```diff
@@ -283,3 +287,106 @@ def size_to_color(size):
     norm = mpl_colors.Normalize(vmin=0, vmax=784)
     rgba = cmap(norm(size))
     return mpl_colors.rgb2hex(rgba)
+
+
+def calculate_true_positives(
+    actual: torch.Tensor, predicted: torch.Tensor, label: int
+) -> tuple[float, float, float]:
+    """Calculate true positives, false positives and false negatives of a specific label
+
+    Parameters
+    ----------
+    actual : torch.Tensor
+        true labels
+    predicted : torch.Tensor
+        predicted labels
+    label : int
+        target label to calculate metrics
+
+    Returns
+    -------
+    tuple[float, float, float]
+        true positives, false positives, false negatives
+    """
+    true_positives = np.sum((actual == label) & (predicted == label))
+    false_positives = np.sum((actual != label) & (predicted == label))
+    false_negatives = np.sum((predicted != label) & (actual == label))
+
+    return true_positives, false_positives, false_negatives
+
+
+def f1(actual: torch.Tensor, predicted: torch.Tensor, label: int) -> float:
```

Comment:
f1 is already implemented in torcheval, it is still marked as unstable but very actively supported by Meta. There is also sklearn that proposed a very robust implementation (need to transform Tensor into ndarray).
Do you have a specific reason to re-implement existing code?

Reply:
I was using torchmetrics initially and removed it to reduce dependencies

Reply:
ok, we will have to discuss if we rely on external libs for evaluation metrics. The balance is always hard to find between avoiding dependencies and recoding everything, that is adding bugs.